Simple Inventory Management System/Simple Inventory Management System/Inventory .cs

```
@@ -0,0 +1,126 @@
+﻿
+namespace Simple_Inventory_Management_System
+{
+    public class Inventory
+    {
+        public List<Product> Products { get; private set; }
+        public Inventory(List<Product> products)
+        {
+            this.Products = products;
+        }
+        public Inventory()
+        {
+            this.Products = new List<Product>();
+        }
+        public bool Add(Product product)
+        {
+            if (IsNotValid(product))
+                return false;
+
+            var existingProduct = CheckIfExist(product);
+
+            if (existingProduct != null)
```

if the product exists then we shouldn't update it, we should throw an exception indicating that this product already exists. It would be confusing to update the product when the user asks explicitly to add it. This gives the user the option to recheck his input for any errors.